src/client/components/Threads.tsx

```diff
@@ -288,12 +295,35 @@ export const OpenThreadContext = createContext<{
   onOpenThread: () => {},
 });
 
+const LoadMoreThreadsContext = createContext<{
+  fetchMore: (howMany: number) => Promise<void>;
+  loading: boolean;
+  hasMore: boolean;
+}>({
+  fetchMore: async (_: number) => {},
+  loading: true,
+  hasMore: false,
+});
+
 function ThreadsScrollContainer(props: betaV2.ScrollContainerProps) {
+  const { fetchMore, loading, hasMore } = useContext(LoadMoreThreadsContext);
+  const onScrollToEdge = useCallback(
+    (edge: string) => {
```

Comment:
It is a bit annoying you have to do that (context + rewriting the `onScrollToEdge`
I wonder if we could make it simpler.